src/wp-admin/network/users.php

```diff
@@ -107,12 +117,27 @@
 
 							case 'notspam':
 								$user = get_userdata( $user_id );
+								if ( is_super_admin( $user->ID ) ) {
+									wp_die(
+										sprintf(
+										/* translators: %s: User login. */
+											__( 'Warning! User cannot be modified. The user %s is a network administrator.' ),
+											esc_html( $user->user_login )
+										)
+									);
+								}
 
 								$userfunction = 'all_notspam';
-								$blogs        = get_blogs_of_user( $user_id, true );
 
-								foreach ( (array) $blogs as $details ) {
-									update_blog_status( $details->userblog_id, 'spam', '0' );
+								/** This filter is documented in wp-admin/network/users.php#L95 */
+								if ( apply_filters( 'handle_network_allusers_update_blog_status', false ) ) {
+									$blogs = get_blogs_of_user( $user_id, true );
+
+									foreach ( $blogs as $details ) {
+										if ( ! is_main_site( $details->userblog_id ) ) { // Main site is never a spam!
+											update_blog_status( $details->userblog_id, 'spam', '0' );
+										}
```

Comment:
What if we have multiple networks setup here? Each network would have it's own main site. 

```suggestion
										if ( ! is_main_site( $details->userblog_id ) && $details->site_id === get_current_network_id() ) { // Main site is never a spam and part of the current network. 
											update_blog_status( $details->userblog_id, 'spam', '0' );
										}
```